send_stream_test.go

```diff
@@ -318,6 +319,7 @@ var _ = Describe("Send Stream", func() {
 			Expect(str.Context().Done()).ToNot(BeClosed())
 			Expect(str.Close()).To(Succeed())
 			Expect(str.Context().Done()).To(BeClosed())
+			Expect(context.Cause(str.Context())).To(MatchError(context.Canceled))
```

Comment:
Is there any way to return `nil` here?

Reply:
Calling a context's cancel func with `nil` will result in the context being cancelled with `context.Canceled`.

This from the [`WithCancelCause`](https://pkg.go.dev/context#WithCancelCause) docs:
> Calling cancel with a non-nil error (the "cause") records that error in ctx; it can then be retrieved using Cause(ctx). Calling cancel with nil sets the cause to Canceled.

Or maybe I misunderstood what you are asking.

Reply:
Btw, the same thing happens when Closing the a `quic.Connection` by closing the `Listener` (which, unlike a `Connection`, doesn't accept an error argument to `Close`). There, the context also resolves to `context.Canceled` which, I think, is the same error that you get when on `conn.AcceptStream()` when calling `ln.Close()`

Reply:
What I meant to say is that we do call the context cancel fun with `nil`, see https://github.com/quic-go/quic-go/pull/3970/files#diff-0683df4767cffeaa4b1127b3e6027900d06dac7973a80454782e810871af742eR369, but the context API treats that as `context.Canceled`

Reply:
Makes sense, thanks for the explanation @fholzer!